src/pages/productpage/finalProducts/finalProducts.js

```diff
@@ -26,11 +26,16 @@ export default function FinalProducts(){
    
     return (
       <div >
-        {error && <h1>{error}</h1>}
+        {loader && <div class="loader"></div>}
         <div className="products">
           <div className="product-grid-container">
-        {newProducts &&
-          newProducts.map((product) => {
+         { newProducts.filter(product => {
+                if(searchTerm === ""){
+                  return product
+                }else if(product.title.toLowerCase().includes(searchTerm.toLowerCase())){
+                  return product
+                }
+              }).map((product) => {
```

Comment:
extract this into a function or variable and then map, as the code is not readable